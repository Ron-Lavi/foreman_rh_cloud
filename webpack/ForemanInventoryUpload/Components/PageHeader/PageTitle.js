import React, { useState } from 'react';
import {
  Dropdown,
  DropdownItem,
  KebabToggle,
  DropdownPosition,
} from '@patternfly/react-core';
import {
  INVENTORY_PAGE_TITLE,
  ACTIONS_HISTORY_BUTTON_TEXT,
  DOCS_BUTTON_TEXT,
  CLOUD_PING_TITLE,
} from '../../ForemanInventoryConstants';
import {
  getActionsHistoryUrl,
  getInventoryDocsUrl,
} from '../../ForemanInventoryHelpers';
import CloudPingModal from './components/CloudPingModal';

const PageTitle = () => {
  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
  const [showPingModal, setPingModal] = useState(false);
  const togglePingModal = () => setPingModal(v => !v);
  const dropdownItems = [
    <DropdownItem
      key="tasks-history-button"
      href={getActionsHistoryUrl()}
      target="_blank"
      rel="noopener noreferrer"
    >
      {ACTIONS_HISTORY_BUTTON_TEXT}
    </DropdownItem>,
    <DropdownItem
      key="inventory-documentation-button"
      href={getInventoryDocsUrl()}
      target="_blank"
      rel="noopener noreferrer"
    >
      {DOCS_BUTTON_TEXT}
    </DropdownItem>,
    <DropdownItem key="cloud-ping" onClick={togglePingModal}>
      {CLOUD_PING_TITLE}
    </DropdownItem>,
  ];
  return (
    <div className="row form-group inventory-upload-header-title">
      <h1 className="col-md-8">{INVENTORY_PAGE_TITLE}</h1>
      <Dropdown
        className="title-dropdown"
        onSelect={() => setIsDropdownOpen(false)}
        toggle={<KebabToggle onToggle={isOpen => setIsDropdownOpen(isOpen)} />}
        isOpen={isDropdownOpen}
        isPlain
        dropdownItems={dropdownItems}
        position={DropdownPosition.right}
      />
      <CloudPingModal
        isOpen={showPingModal}
        toggle={togglePingModal}
        title={CLOUD_PING_TITLE}
      />
    </div>
  );
};
export default PageTitle;
