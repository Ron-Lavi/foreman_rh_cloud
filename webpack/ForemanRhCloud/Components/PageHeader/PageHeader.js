import React from 'react';
import { Grid } from 'patternfly-react';
import AutoUploadSwitcher from '../AutoUploadSwitcher';

const PageHeader = () => (
  <Grid.Row>
    <Grid.Col xs={4} xsOffset={8}>
      <AutoUploadSwitcher />
    </Grid.Col>
  </Grid.Row>
);

export default PageHeader;
