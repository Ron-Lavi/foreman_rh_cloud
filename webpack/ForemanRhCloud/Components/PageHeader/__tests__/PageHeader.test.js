import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import PageHeader from '../PageHeader';

const fixtures = {
  'render without Props': {},
  /** fixtures, props for the component */
};

describe('PageHeader', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(PageHeader, fixtures));
});
