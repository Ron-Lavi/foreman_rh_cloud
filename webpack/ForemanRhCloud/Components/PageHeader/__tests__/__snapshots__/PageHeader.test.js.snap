// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader rendering render without Props 1`] = `
<Row
  bsClass="row"
  componentClass="div"
>
  <Col
    bsClass="col"
    componentClass="div"
    xs={4}
    xsOffset={8}
  >
    <Connect(AutoUploadSwitcher) />
  </Col>
</Row>
`;
